core/src/main/java/org/kiwix/kiwixmobile/core/main/AddNoteDialog.java

```diff
@@ -81,23 +81,19 @@
   TextView addNoteTextView; // Displays article title
   @BindView(R2.id.add_note_edit_text)
   EditText addNoteEditText; // Displays the note text
-
+  @Inject SharedPreferenceUtil sharedPreferenceUtil;
+  @Inject ZimReaderContainer zimReaderContainer;
   private Unbinder unbinder;
-
   private String zimFileName;
   private String zimFileTitle;
   private String articleTitle;
   // Corresponds to "ArticleUrl" of "{External Storage}/Kiwix/Notes/ZimFileName/ArticleUrl.txt"
   private String articleNotefileName;
+  // Keeps track of state of the note (whether edited since last save)
```

Comment:
this is in the wrong location now, please revert all the changes except
`addNoteEditText.setSelection(addNoteEditText.getText().length() - 1);`

Reply:
Done